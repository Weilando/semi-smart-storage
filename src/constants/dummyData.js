const DummyItemList = [
  {id: 0, name: "Sparkling water"},
  {id: 1, name: "Apple juice"},
  {id: 2, name: "Craft beer"},
  {id: 3, name: "Milk"},
  {id: 5, name: "Coke"}
];

const DummyItemListShort = [
  {id: 0, name: "Sparkling water"},
  {id: 1, name: "Apple juice"},
  {id: 3, name: "Milk"}
];

const DummyStorageContent = [
  {id: 0, item: "Milk", unit: "1L", quantity: 2.7},
  {id: 1, item: "Coke", unit: "0.33L", quantity: 5},
  {id: 2, item: "Apple juice", unit: "0.5L", quantity: 2},
  {id: 4, item: "Craft Beer", unit: "0.33L", quantity: 6},
  {id: 5, item: "Sparkling water", unit: "0.75L", quantity: 21}
];

const DummyStorageContentShort = [
  {id: 0, item: "Milk", unit: "1L", quantity: 2.7},
  {id: 2, item: "Sparkling water", unit: "0.75L", quantity: 21}
];

const DummyStorageList = [
  {id: 0, name: 'Fridge'},
  {id: 1, name: 'Basement'},
  {id: 3, name: 'Garage'},
];

const DummyStorageListShort = [
  {id: 0, name: 'Basement'},
  {id: 3, name: 'Fridge'},
];

const DummyUnitList = [
  {id: 0, name: "0.33L"},
  {id: 1, name: "0.5L"},
  {id: 2, name: "0.75L"},
  {id: 3, name: "1.0L"}
];

const DummyUnitListShort = [
  {id: 0, name: "0.33L"},
  {id: 2, name: "0.75L"},
  {id: 3, name: "1.0L"}
];

export { DummyItemList, DummyStorageContent, DummyStorageList, DummyUnitList };
export { DummyItemListShort, DummyStorageContentShort, DummyStorageListShort, DummyUnitListShort };
