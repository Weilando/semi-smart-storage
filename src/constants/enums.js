// api-operations
const AddMode = {
  ITEM_LIST: 0, STORAGE_LIST: 1, STORAGE_CONTENT: 2, UNIT_LIST: 3
}

const DeleteMode = {
  ITEM_LIST: 0, STORAGE_LIST: 1, STORAGE_CONTENT: 2, UNIT_LIST: 3
}

const FetchMode = {
  ITEM_LIST: 0, STORAGE_LIST: 1, STORAGE_CONTENT: 2, UNIT_LIST: 3
}

const UpdateMode = {
  ITEM_LIST: 0,
  STORAGE_LIST: 1,
  STORAGE_CONTENT_DECREMENT: 2,
  STORAGE_CONTENT_INCREMENT: 3,
  STORAGE_CONTENT_EDIT: 4,
  UNIT_LIST: 5
}

// modal-modes
const ModalMode = {
  ITEM: "item",
  STORAGE: "storage",
  UNIT: "unit"
}

export { AddMode, DeleteMode, FetchMode, UpdateMode, ModalMode };
