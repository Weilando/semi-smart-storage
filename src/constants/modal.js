let ModalMode = {
  ITEM: "item",
  STORAGE: "storage",
  UNIT: "unit"
}

export { ModalMode }
