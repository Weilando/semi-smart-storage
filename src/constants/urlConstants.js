const BASEURL = "http://raspberrypi/api/";
const ADD = "add.php";
const DELETE = "delete.php";
const GET = "get.php";
const UPDATE = "update.php";

export { BASEURL };
export { ADD, DELETE, GET, UPDATE };
