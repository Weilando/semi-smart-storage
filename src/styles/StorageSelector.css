div.StorageSelector {
  display: inline-block;
  width: 23%;
  height: 98%;
  border: 1px solid orange;
  margin-right: 0.5%;
  margin-left: 0;
  padding-left: 0.2%;
  padding-right: 0.2%;
}

ul.StorageSelector {
  list-style-type: none;
  text-align: left;
  padding: 0;
  margin: 0;
}

li.StorageSelector {
  min-height: 2em;
  background-color: lightblue;
  margin: 0.2em;
  padding: 0.2em;
}

li.StorageSelector:hover {
  background-color: lightgreen;
}

input[type="text"].StorageSelector {
  background-color: transparent;
  width: 70%;
  border: none;
}

input[type="submit"].entry {
  float: right;
  margin-right: 0;
}
