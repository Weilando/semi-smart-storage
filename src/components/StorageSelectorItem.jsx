import React from 'react';
import PropTypes from 'prop-types';
import '../styles/StorageSelector.css';

function StorageSelectorItem(props) {
  StorageSelectorItem.proptTypes = {
    sName: PropTypes.string.isRequired,
    sId: PropTypes.number.isRequired
  }
  
  return (
    <li className="StorageSelector" onClick={function() { alert('click '.concat(props.sId)); }}>
      {props.sName}
    </li>
  );
}

export default StorageSelectorItem;
