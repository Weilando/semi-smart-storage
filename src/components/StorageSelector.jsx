import React from 'react';
import '../styles/StorageSelector.css';
import StorageSelectorItem from './StorageSelectorItem';
import StorageSelectorNewItem from './StorageSelectorNewItem';
import { getAllStorages } from '../services/StorageService';

class StorageSelector extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      storageList: [{id: "0", name: "Fridge"}, {id: "1", name: "Basement"}, {id: "2", name: "Garage"}],
    }
  }

  componentDidMount() {
    this.getStorageList();
  }

  getStorageList() {
    // fetch available storages from Raspberry
    getAllStorages().then(storages => {
      this.setState({storageList: storages})
    });
  }

  addStorage(name) {
    alert('add new storage: '.concat(name)); // TODO: Add api call
  }

  unpackStorageList() {
    const keys = [...Array(this.state.storageList.length).keys()]; // Array with keys from 0 to entries.length

    return keys.map((currKey) =>
      <StorageSelectorItem
        sId={this.state.storageList[currKey].id}
        sName={this.state.storageList[currKey].name}
      />
    );
  }

  render() {
    let storageList = this.unpackStorageList();

    return (
      <div className="StorageSelector">
        <ul className="StorageSelector">
          {storageList}
          <StorageSelectorNewItem addStorage={this.addStorage}/>
        </ul>
      </div>
    );
  }
}

export default StorageSelector;
