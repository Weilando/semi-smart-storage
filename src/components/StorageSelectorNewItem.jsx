import React from 'react';
import PropTypes from 'prop-types';
import '../styles/StorageSelector.css';

class StorageSelectorNewItem extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      content: '',
    }

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  static propTypes = {
    addStorage: PropTypes.func.isRequired,
  }

  handleChange(event) {
    this.setState({content: event.target.value});
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.addStorage(this.state.content);
    this.setState({content: ''});
  }

  render() {
    return (
      <li className="StorageSelector">
        <form onSubmit={this.handleSubmit}>
          <input
            type="text"
            className="StorageSelector"
            value={this.state.content}
            onChange={this.handleChange}
          />
          <input
            type="submit"
            className="StorageSelector"
            value="Add"
            disabled={this.state.content === ''}
          />
        </form>
      </li>
    );
  }

}

export default StorageSelectorNewItem;
