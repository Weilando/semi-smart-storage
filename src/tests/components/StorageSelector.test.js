import React from 'react';
import renderer from 'react-test-renderer';
import StorageSelector from '../../components/StorageSelector';
import StorageSelectorItem from '../../components/StorageSelectorItem';
import StorageSelectorNewItem from '../../components/StorageSelectorNewItem';
import { DummyStorageListShort } from '../../constants/dummyData';

describe('snapshot tests for StorageSelector component', () => {
  it('should render initial StorageSelector component correctly', () => {
    const addAction = jest.fn();
    const deleteAction = jest.fn();
    const switchStorageAction = jest.fn();
    const updateAction = jest.fn();

    const tree = renderer.create(
      <StorageSelector
        currentStorage={DummyStorageListShort[0]}
        storageList={DummyStorageListShort}
        addAction={addAction}
        deleteAction={deleteAction}
        switchAction={switchStorageAction}
        updateAction={updateAction}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })
});

describe('snapshot tests for StorageSelectorItem component', () => {
  it('should render initial active StorageSelectorItem component correctly', () => {
    const storage = {id: 0, name: 'Fridge'};
    const editAction = jest.fn();
    const showAction = jest.fn();

    const tree = renderer.create(
      <StorageSelectorItem
        storage={storage}
        editAction={editAction}
        showAction={showAction}
        active={true}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })

  it('should render initial inactive StorageSelectorItem component correctly', () => {
    const storage = {id: 0, name: 'Fridge'};
    const editAction = jest.fn();
    const showAction = jest.fn();

    const tree = renderer.create(
      <StorageSelectorItem
        storage={storage}
        editAction={editAction}
        showAction={showAction}
        active={false}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })
});

describe('snapshot tests for StorageSelectorNewItem component', () => {
  it('should render initial StorageSelectorNewItem component correctly', () => {
    const tree = renderer.create(
      <StorageSelectorNewItem
        addStorage={jest.fn()}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })
});
