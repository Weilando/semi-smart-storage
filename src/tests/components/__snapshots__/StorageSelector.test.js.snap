// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for StorageSelector component should render initial StorageSelector component correctly 1`] = `
<div
  className="StorageSelector"
>
  <ul
    className="StorageSelector"
  >
    <li
      className="StorageSelectorActive"
    >
      <button
        className="Block"
        onClick={[Function]}
      >
        <p
          className="Active"
        >
          Basement
        </p>
      </button>
      <button
        className="Round"
        onClick={[Function]}
      >
        i
      </button>
    </li>
    <li
      className="StorageSelector"
    >
      <button
        className="Block"
        onClick={[Function]}
      >
        <p
          className=""
        >
          Fridge
        </p>
      </button>
      <button
        className="Round"
        onClick={[Function]}
      >
        i
      </button>
    </li>
    <li
      className="StorageSelector"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="StorageSelector"
          onChange={[Function]}
          type="text"
          value=""
        />
        <input
          className="StorageSelector"
          disabled={true}
          type="submit"
          value="Add"
        />
      </form>
    </li>
  </ul>
  <div
    className="modal hide"
  >
    <div
      className="modal-main"
    >
      <span
        className="close"
        onClick={[Function]}
      >
        Ã—
      </span>
      <h1>
        Settings for 
        storage
         "
        Fridge
        ".
      </h1>
      <h2>
        Update name
      </h2>
      <form
        className="Modal"
        onSubmit={[Function]}
      >
        <input
          className="Modal"
          onChange={[Function]}
          type="text"
          value=""
        />
        <br />
        <input
          className="Modal"
          disabled={false}
          type="submit"
          value="Update name"
        />
      </form>
      <h2>
        Delete 
        storage
      </h2>
      <p>
        To delete the storage and all its content, click the button below. Items and units are not deleted.
      </p>
      <button
        className="Modal"
        onClick={[Function]}
      >
        Delete 
        storage
      </button>
    </div>
  </div>
</div>
`;

exports[`snapshot tests for StorageSelectorItem component should render initial active StorageSelectorItem component correctly 1`] = `
<li
  className="StorageSelectorActive"
>
  <button
    className="Block"
    onClick={[Function]}
  >
    <p
      className="Active"
    >
      Fridge
    </p>
  </button>
  <button
    className="Round"
    onClick={[Function]}
  >
    i
  </button>
</li>
`;

exports[`snapshot tests for StorageSelectorItem component should render initial inactive StorageSelectorItem component correctly 1`] = `
<li
  className="StorageSelector"
>
  <button
    className="Block"
    onClick={[Function]}
  >
    <p
      className=""
    >
      Fridge
    </p>
  </button>
  <button
    className="Round"
    onClick={[Function]}
  >
    i
  </button>
</li>
`;

exports[`snapshot tests for StorageSelectorNewItem component should render initial StorageSelectorNewItem component correctly 1`] = `
<li
  className="StorageSelector"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="StorageSelector"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      className="StorageSelector"
      disabled={true}
      type="submit"
      value="Add"
    />
  </form>
</li>
`;
